src/servers/main.cc

```diff
@@ -346,15 +472,22 @@ Parse(nvidia::inferenceserver::InferenceServer* server, int argc, char** argv)
 
   bool exit_on_error = exit_on_failed_init_;
 
-  bool allow_http = true;
-  bool allow_grpc = true;
   bool allow_metrics = true;
   bool allow_gpu_metrics = true;
   int32_t http_port = http_port_;
   int32_t grpc_port = grpc_port_;
   int32_t metrics_port = metrics_port_;
   int32_t http_thread_cnt = http_thread_cnt_;
 
+  int32_t http_status_port = -1;
```

Comment:
To get the correct in-order command-line parsing behavior you need to initialize these http_port_ or grpc_port_. Then in the case statements update them appropriately. For example if --http-port=-1 is parsed then you need to set all 4 http port variables to -1. After this first initialization you shouldn't use http_port_ or grpc_port_ anywhere else.